### PR TITLE
[New Rule] Netcon through XDG Autostart Entry

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
Research related to this PR is conducted and documented in https://github.com/elastic/ia-trade-team/issues/374. This issue also contains the query + validation aspect.

This PR:

- Adds a new rule to detect network connections initiated by a XDG Autostart entry on Linux XFCE/GNOME desktop distributions.

- Rule coverage will be increased when new Linux distributions are created, and this technique is tested.

No FPs last 30d my stack, no FPs last 7d in telemetry.

### CODE REVIEW

**Comment by terrancedejesus** [comment]
> No FPs last 30d my stack, no FPs last 7d in telemetry.

Does this indicate we had FPs after the last 7d in telemetry? If so, is the amount extensive?

File: rules/linux/persistence_xdg_autostart_netcon.toml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+Detects network connections initiated through XDG autostart entries for GNOME and XFCE-based Linux distributions. XDG
+Autostart entries can be used to execute arbitrary commands or scripts when a user logs in. This rule helps to identify
+potential malicious activity where an attacker may have modified XDG autostart scripts to establish persistence on the
+system.
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*", "endgame-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Network connections Initiated Through XDG Autostart Entry"
```
Comment (terrancedejesus) [inline rules/linux/persistence_xdg_autostart_netcon.toml:19]:
```suggestion
name = "Network Connections Initiated Through XDG Autostart Entry"
```

**Review by terrancedejesus — APPROVED** [review]
Small suggestion, but LGTM!

File: rules/linux/persistence_xdg_autostart_netcon.toml [hunk 2]
```diff
@@ -0,0 +1,100 @@
+[metadata]
+creation_date = "2024/06/03"
+integration = ["endpoint"]
+maturity = "production"
+updated_date = "2024/06/03"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detects network connections initiated through XDG autostart entries for GNOME and XFCE-based Linux distributions. XDG
```
Comment (Mikaayenson) [inline rules/linux/persistence_xdg_autostart_netcon.toml:10]: nit: 
```suggestion
Detects network connections initiated through Cross-Desktop Group (XDG) autostart entries for GNOME and XFCE-based Linux distributions. XDG
```

**Review by Mikaayenson — APPROVED** [review]
LGTM. Wasn't sure if you tested against the endgame data source but it may be good to double check just in case.

**Comment by Aegrah** [comment]
@terrancedejesus No, some sequences are more compute-intense than others, so some sequences I can easily run for 14+ days, while others start taking up a lot of resources at 7+ days in telemetry. This one was tested in telemetry with 7 days of data, and locally with 30 days.

**Comment by Aegrah** [comment]
@Mikaayenson you are right, the sensor was not running on the test VM's so therefore I did not validate. `process.entity_id` is indeed not supported by Endgame sensor, so I removed it. Ty.

Added references.

**Pull request merged**